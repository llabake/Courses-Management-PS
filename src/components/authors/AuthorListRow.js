/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AuthorListRow = ({ author, onDelete }) => {
  const deleteAuthor = () => (
    onDelete(author.id)
  );
  return (
    <tr>
      <td><Link to={`/author/${author.id}`}>{author.id}</Link></td>
      <td>
        {author.firstName} 
        {' '}
        {author.lastName}
      </td>
      <td
        className="btn btn-xs btn-danger"
        onClick={deleteAuthor}
      >
        Delete
      </td>
    </tr>
  );
};

AuthorListRow.propTypes = {
  author: PropTypes.object.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default AuthorListRow;
