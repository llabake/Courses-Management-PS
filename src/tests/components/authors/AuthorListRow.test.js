import React from 'react';
import { shallow } from 'enzyme';
import AuthorListRow from "../../../components/authors/AuthorListRow";
import authorMock from "../../mocks/authorMock";

describe('<AuthorListRow />', () => {
  const props = {
    onDelete: jest.fn(),
    author: authorMock.createdAuthorWithId,
  };
  const wrapper = shallow(<AuthorListRow {...props} />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('deletes an author', () => {
    wrapper.find('.btn').simulate('click');
    expect(props.onDelete.mock.calls.length).toEqual(1);
  });
});
