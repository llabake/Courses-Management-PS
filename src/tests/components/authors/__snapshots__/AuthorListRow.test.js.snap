// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorListRow /> renders correctly 1`] = `
<tr>
  <td>
    <Link
      replace={false}
      to="/author/sardaunan-happygirl"
    >
      sardaunan-happygirl
    </Link>
  </td>
  <td>
    Sardaunan
     
    Happygirl
  </td>
  <td
    className="btn btn-xs btn-danger"
    onClick={[Function]}
  >
    Delete
  </td>
</tr>
`;
